viewing_party/party.py

```diff
@@ -1,23 +1,123 @@
 # ------------- WAVE 1 --------------------
 
+from hashlib import new
+from re import M
+from collections import Counter
+from viewing_party.helper_functions import *
+
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1, USER_DATA_3
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        new_movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data = {
+        "watched": [movie]
+    }             
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {
+        "watchlist": [movie]
+    }             
+    return user_data
+
+def watch_movie(user_data, title):    
+    new_watchlist = []
+    # iterates over watchlist. moves value in arg title from watchlist to 
+    # watched if found. generates a new watchlist comprised of unwatched movies 
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+        else:
+            new_watchlist.append(movie)
+    user_data["watchlist"] = new_watchlist
+    return user_data

```

Comment:
Instead of creating `new_watchlist` and appending movies to it and then using new_watchlist to overwrite the existing watchlist on line 42, you could append the movie to user_data["watched"].append(movie) like you do on line 39. 

Then while you're still in that iteration of the loop, you can use the remove() function to remove the movie from watchlist. 

```
user_data["watched"].append(movie)
user_data["watchlist"].remove(movie)
```